### ISSUE TITLE
feat(source-contentful): Ensure that the schema generated for missing fields is correct

### ISSUE THREAD

**Issue by weaversam8** [gridsome/gridsome#874]
## Summary

See #864 and #866. See [this comment](https://github.com/gridsome/gridsome/pull/866#issuecomment-564532105).

**Comment by weaversam8** [comment]
Adding some notes for whoever implements this later. (Even if it might be me.)

I think the basic way that this will be implemented is:

- modify the [`getContentTypes( ... )`](https://github.com/gridsome/gridsome/blob/55dc559e9ec7072042cb512b5d5defb7ef7360cb/packages/source-contentful/index.js#L36) function in the `ContentfulSource` class
  - after the called to `actions.addCollection( ... )` and `actions.addSchemaResolvers( ... )`, we'll probably want to make a call to [`actions.addSchemaTypes( types )`](https://gridsome.org/docs/schema-api/#addschematypestypes) with the constructed schema object
  - honestly, we should probably start building the schema object via [this `for` loop](https://github.com/gridsome/gridsome/blob/55dc559e9ec7072042cb512b5d5defb7ef7360cb/packages/source-contentful/index.js#L46)
  - for reference, we could also store the schema object [on this line](https://github.com/gridsome/gridsome/blob/55dc559e9ec7072042cb512b5d5defb7ef7360cb/packages/source-contentful/index.js#L57) but that's probably not required for anything later.

I haven't actually tested any of this, but I think that's probably how we'd implement this.

**Comment by hjvedvik** [comment]
@weaversam8 Yes, the `getContentTypes()` method is a good place to generate the schema types for each collection. And I think the content types from Contentful has all the necessary info to generate them. We will implement it eventually, but feel free to open a PR for it if you got the time to look into it :)
